src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -340,4 +349,43 @@ class EntityClient (requestContext: RequestContext, modelSchema: ModelSchema)(im
       }
     }
   }
+
+  def importPFB(pipeline: WithTransformerConcatenation[HttpRequest, Future[HttpResponse]],
+                workspaceNamespace: String, workspaceName: String, pfbRequest: PfbImportRequest): Future[PerRequestMessage] = {
+
+    def callArrow(url: String) = {
+      pipeline { Post(avroToRawlsURL, PfbImportRequest(url)) }
+    }
+
+    def callRawls(entity: HttpEntity) = {
+      pipeline {
+        Post(FireCloudDirectiveUtils.encodeUri(Rawls.entityPathFromWorkspace(workspaceNamespace, workspaceName) + "/batchUpsert"), entity)
+      }
+    }
+
+    val pfbUrl = new URL(pfbRequest.url)
+    val acceptableProtocols = Seq("https") //for when we inevitably change our mind and need to support others
+
+    if (!acceptableProtocols.contains(pfbUrl.getProtocol)) {
+      Future.successful(RequestCompleteWithErrorReport(StatusCodes.BadRequest, "Invalid URL protocol: must be https only"))
+    } else {
+
+      val futurePerRequestMessage = for {
+        arrowResponse <- callArrow(pfbUrl.toString)
+        rawlsJsonEntity <- arrowResponse.status match {
+          case OK => Future.successful(arrowResponse.entity)
+          case _ => Future.failed(new FireCloudException(arrowResponse.status.value + ": " + arrowResponse.entity.asString))
```

Comment:
suggest using:
```
new FireCloudExceptionWithErrorReport(ErrorReport(arrowResponse.status, arrowResponse.entity.asString))
```
since it doesn't rely on a delimited string. Same applies to the `rawlsResponse.status match` block below.

Reply:
I don't want to do this at the moment because, with the way 404 from Rawls is being communicated back up, making this a `FireCloudExceptionWithErrorReport` will cause the status code from Arrow to propagate up which I don't think is universally the right thing.